packages/shared/lib/clients/orchestrator.ts

```diff
@@ -590,6 +591,7 @@ export class Orchestrator {
                     const syncConfig = await getSyncConfigBySyncId(syncId);
                     for (const model of syncConfig?.models || []) {
                         const del = await recordsService.deleteRecordsBySyncId({ syncId, connectionId, model });
+                        await recordsService.deleteRecordCount({ connectionId, environmentId, model });
```

Comment:
I would call `deleteRecordCount` directly inside `deleteRecordsBySyncId` since it always need to happen when deleting records